.github/workflows/windows-ci.yml

```diff
@@ -0,0 +1,91 @@
+---
+name: Windows build and test
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '*'
+  release:
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: windows-2019
+    strategy:
+      fail-fast: false
+      matrix:
+        include: [
+          { arch: x86_64, msystem: mingw64, tests: none },
+          { arch: x86_64, msystem: mingw64, tests: main },
+          { arch: i686, msystem: mingw32, tests: main },
+          { arch: i686, msystem: mingw32, tests: none },
+        ]
```

Comment:
Our discussion's been a little disjointed and I think there are some disconnects in what we've each been trying to communicate, so maybe it's good to summarize here. I think this should be
```
        include: [
          { arch: x86_64, msystem: mingw64, debug: false },
          { arch: x86_64, msystem: mingw64, debug: true },
          { arch: i686, msystem: mingw32, debug: false },
          { arch: i686, msystem: mingw32, debug: true },
        ]
```
and then down below, we have
```
[[ ${{ matrix.debug }} == "true" ]] && tests ="none" suffix="" || tests = main dbg_flags = "--enable-debug" suffix="-dbg"
```
But the real logic looks to me to be
```
matrix:
   arch: [x86_64, i686]
   debug: [true, false]
```
and then
```
[[ ${{ matrix.debug }} == "true" ]] && tests ="none" suffix="" || tests = main dbg_flags = "--enable-debug" suffix="-dbg"
[[ ${{ matrix.arch }} == "x86_64" ]] && msystem = "mingw64" || msystem = "mingw32
```
Later, we can add a new arch for MSVC, as such.
```
   arch: [x86_64, i686, win64]
```
But I would even name `arch` to `platform` to make it clearer what it means
```
```
matrix:
   platform: [x86_64, i686]
   debug: [true, false]
```

Reply:
I'm a bit exasperated on this PR, so feel free to push the changes that you feel provide the correct coverage.